### DISCUSSION TITLE
How can I sort items by custom value in Prisma? (trying to implement “sort by hot” like on Hacker News or Reddit)

### DISCUSSION THREAD

**Discussion by lumenwrites** [prisma/prisma#8043] [Q&A] [answered]
I'm trying to implement "sort by hot" similar to what HackerNews/Reddit do, as described [here](https://medium.com/hacking-and-gonzo/how-hacker-news-ranking-algorithm-works-1d9b0cf2c08d).

I want to calculate the rank of a post like so:
```ts
    const age = new Date() - createdAt
    const gravity = 1.8
    const rankScore = (score - 1) / (age + 2) ** gravity
```
The `Post` model has a `score` value - how many upvotes the post has received, and `createdAt` - the time it has been created. The higher the `rankScore` is, the closer I want the post to be to the top of the list. So I want to take the `score` and `createdAt` values, use them to calculate `rankScore`, and then do `orderBy` by that.

What is the right way to implement this in Prisma?

Right now I'm just returning the list of most upvoted posts like so:
```ts
    prisma.post.findMany({
      where: {
        authorId: authorId,
        published: args.published || undefined,
      },
      orderBy: [{ score: "desc" }],
    })
```
So it's easy to rank posts by a field that already exists on the model, but I need to calculate it in realtime, as the page loads, because `rankScore` depends on post's age, and it wouldn't make sense to store something like this in the db.

**Comment by ryands17** [comment, marked as answer]
@lumenwrites 👋 

Currently a [raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) would be the only way to do this. I would suggest creating a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for this with the above use case so that we can look into this 🙂

**Reply by asobirov** [reply]
Hey, was wondering if this still has to be done through a raw query?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
